tests/js/session-tests.js

```diff
@@ -1033,6 +1039,116 @@ module.exports = {
         });
     },
 
+    async testClientResetAtOpen() {
+        // FIXME: try to enable for React Native
+        if (!isNodeProcess) {
+            return;
+        }
+
+        if (!global.testAdminUserInfo) {
+            throw new Error("Test requires an admin user");
+        }
+
+        let called = false;
+        const credentials = Realm.Sync.Credentials.usernamePassword(global.testAdminUserInfo.username, global.testAdminUserInfo.password);
+        let user = await Realm.Sync.User.login('http://127.0.0.1:9080', credentials);
+        var realm;
+        const config = user.createConfiguration({ sync: { url: 'realm://127.0.0.1:9080/~/myrealm' } });
+        config.schema = [schemas.IntOnly];
+        config.sync.clientResyncMode = 'manual';
+        config.sync.fullSynchronization = true;
+        config.sync.error = (sender, error) => {
+            called = true;
+            try {
+                TestCase.assertEqual(error.name, 'ClientReset');
+                TestCase.assertDefined(error.config);
+                TestCase.assertNotEqual(error.config.path, '');
+                const path = realm.path;
+                realm.close();
+                Realm.Sync.initiateClientReset(path);
+                // open Realm with error.config, and copy required objects a Realm at `path`
+            }
+            catch (e) {
+            }
+        };
+
+        // open, download, create an object, upload and close
+        realm = await Realm.open(config);
+        await realm.syncSession.downloadAllServerChanges();
+        realm.write(() => {
+            realm.create(schemas.IntOnly.name, { intCol: 1 });
+        });
+        await realm.syncSession.uploadAllLocalChanges();
+        realm.close();
+
+        // delete Realm on server
+        let encodedPath = encodeURIComponent(`${user.identity}/myrealm`);
+        let url = new URL(`/realms/files/${encodedPath}`, user.server);
+        let options = {
+            headers: {
+                Authorization: `${user.token}`,
+                'Content-Type': 'application/json',
+            },
+            method: 'DELETE',
+        };
+        await fetch(url.toString(), options);
+
+        // open the Realm again and see it fail
+        return Realm.open(config).then(realm => {
```

Comment:
its an async test why not use await Realm.open here

Reply:
I am testing that `Realm.open()` is failing.

Reply:
That's minor comment on the style. It seems to me it could be cleaner with a `try...catch` and `await Realm.open` here.